checker/src/main/java/org/checkerframework/checker/determinism/DeterminismAnnotatedTypeFactory.java

```diff
@@ -948,6 +951,30 @@ public boolean isSubtype(AnnotationMirror subAnno, AnnotationMirror superAnno) {
                     && AnnotationUtils.areSame(superAnno, POLYDET_UPDET)) {
                 return true;
             }
+            if (AnnotationUtils.areSame(subAnno, POLYDET_NOORDERNONDET)
```

Comment:
Instead of using a giant if statement, could you use a switch statement like the one in org.checkerframework.checker.determinism.DeterminismQualifierPolymorphism#replace.  It would be faster and hopefully easier to read.